pkg/subscriptions/subscriptions-client.js

```diff
@@ -0,0 +1,416 @@
+        console.warn(text, exitDetails);
+        client.subscriptionStatus.status = "not-found";
+        return;
+    }
+
+    /* clear old subscription details */
+    client.subscriptionStatus.products = [];
+
+    var status = parseSingleSubscription(text);
+    client.subscriptionStatus.status = status['Overall Status'];
+
+    needRender();
+
+    /* if refresh was requested, try again - otherwise get details */
+    if (client.subscriptionStatus !== 'Unknown')
+        getSubscriptionDetails();
+}
+
+var gettingStatus = false;
+var getStatusRequested = false;
+/* get subscription summary using 'subscription-manager status'*/
+client.getSubscriptionStatus = function() {
+    if (gettingStatus) {
+        getStatusRequested = true;
+        return;
+    }
+    getStatusRequested = false;
+    gettingStatus = true;
+    /* we need a buffer for 'subscription-manager status' output, since that can fail with a return code != 0
+     * even if we need its output (no valid subscription)
+     */
+    var status_buffer = '';
+    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
+     * without translations and parse the output
+     *
+     * 'subscription-manager status' will only return with exit code 0 if all is well (and subscriptions current)
+     */
+    cockpit.spawn(['subscription-manager', 'status'],
+            { directory: '/', superuser: "try", environ: ['LC_ALL=C'], err: "out" })
+        .stream(function(text) {
+            status_buffer += text;
+            return text.length;
+        }).done(function(text) {
+            gettingStatus = false;
+            processStatusOutput(status_buffer + text, undefined);
+            if (getStatusRequested)
+                client.getSubscriptionStatus();
+        }).fail(function(ex) {
+            gettingStatus = false;
+            processStatusOutput(status_buffer, ex);
+            if (getStatusRequested)
+                client.getSubscriptionStatus();
+        });
+};
+
+/* ideally we could get detailed subscription info via DBus, but we
+ * can't rely on this being present on all systems we work on
+ */
+/*service.subscribe(
+    { path: "/EntitlementStatus",
+      interface: "org.freedesktop.DBUS.Properties",
+      member: "PropertiesChanged"
+    }, function(path, interface, signal, args) {
+    current.text('got properties');
+});*/
+
+// dataChanged: callback that is called with new data when render is needed
+client.init = function(dataChanged) {
+    updateCallback = dataChanged;
+
+    service = cockpit.dbus('com.redhat.SubscriptionManager');
+
+    /* we want to get notified if subscription status of the system changes */
+    service.subscribe(
+        { path: '/EntitlementStatus',
+          interface: 'com.redhat.SubscriptionManager.EntitlementStatus',
+          member: 'entitlement_status_changed'
+        },
+        function(path, dbus_interface, signal, args) {
+            updateTimeout = null;
```

Comment:
window.clearTimeout(updateTimeout) is necessary. Note that passing null to clearTimeout() is totally legal.